src/toeplitzhankel.jl

```diff
@@ -0,0 +1,136 @@
+function partialchol(H::Hankel)
+    # Assumes positive definite
+    σ=Array(eltype(H),0)
+    n=size(H,1)
+    C=Array(eltype(H),n,n)
+    v=[H[:,1];vec(H[end,2:end])]
+    d=diag(H)
+
+    @assert length(v) ≥ 2n-1
+
+    tol=1E-14
+    for k=1:n
+        mx,idx=findmax(d)
+        if mx ≤ tol
+            break
+        end
+        push!(σ,1/mx)
+        @simd for p=1:n
+            @inbounds C[p,k]= v[p+idx-1]             # H[p,idx]
+        end
+        for j=1:k-1,p=1:n
+            @inbounds C[p,k]-=C[p,j]*C[idx,j]*σ[j]
+        end
+        @simd for p=1:n
+            @inbounds d[p]-=C[p,k]^2/mx
+        end
+    end
+    for k=1:length(σ),p=1:n
+        @inbounds C[p,k]*=sqrt(σ[k])
+    end
+    C[:,1:length(σ)]
+end
+
+function toeplitzcholmult(T,C,v)
+    ret=C[:,1].*(T*(C[:,1].*v))
+    for j=2:size(C,2)
+        ret+=C[:,j].*(T*(C[:,j].*v))
+    end
+    ret
+end
+
+
+# Plan a multiply by DL*(T.*H)*DR
+immutable ToeplitzHankelPlan{TT}
+    T::TriangularToeplitz{TT}
+    C::Matrix{TT}   # A cholesky factorization of H: H=CC'
```

Comment:
`toeplitzcholmult` re-allocates every vector `C[:,j]`... perhaps change `C` to a `Vector{Vector{TT}}` (including in `partialchol`)?